### PR TITLE
Add data-auto attribute and init to SignupSumo

### PR DESCRIPTION
**Pull request by kenrick**
Bounty: https://assembly.com/signupsumo/bounties/83

This will allow it be called as a Singleton and also be instantiated

### CODE REVIEW

File: app/assets/javascripts/signups.coffee [hunk 1]
```diff
@@ -24,6 +40,9 @@ class @SignupSumo
       [].unshift.call(arguments, "[SignupSumo v#{NOTIFIER_VERSION}]")
       console.log.apply(console, arguments)
 
+  getDataAttribute: (name) ->
+    document.querySelector("script[#{name}]")?.getAttribute(name)
```
Comment (vanstee) [inline app/assets/javascripts/signups.coffee]: Nice good call.

File: app/assets/javascripts/signups.coffee [hunk 2]
```diff
@@ -1,6 +1,8 @@
 #= require superagent
 
-class @SignupSumo
+class SignupSumo
```
Comment (vanstee) [inline app/assets/javascripts/signups.coffee]: Does this still get set on `window` if you remove the `@`?

File: app/assets/javascripts/signups.coffee [hunk 3]
```diff
@@ -38,24 +57,29 @@ class @SignupSumo
...
+      return superagent
+        .post(endpoint)
+        .send(data)
+    else
+      throw 'Token required to send emails. Do `SignupSumo.setToken(TOKEN)` to set it'
 
   validateEmailFormat: (email) ->
     !!email.match(EMAIL_REGEX)
 
+window.SignupSumo = SignupSumo
```
Comment (kenrick) [inline app/assets/javascripts/signups.coffee]: I'm setting it explicitly here. @vanstee

Comment (vanstee) [inline app/assets/javascripts/signups.coffee]: Ah cool I like that better. Nice.

**Pull request merged**